tests/_models/test_package.py

```diff
@@ -96,58 +100,55 @@ def test_dist_package_as_dict() -> None:
 )
 def test_dist_package_licenses(mocked_metadata: Mock, expected_output: str, monkeypatch: pytest.MonkeyPatch) -> None:
     monkeypatch.setattr("pipdeptree._models.package.metadata", lambda _: mocked_metadata)
-    dist = DistPackage(Mock(project_name="a", key="a"))
+    dist = DistPackage(Mock(project_name="a", metadata={"Name": "a"}))
     licenses_str = dist.licenses()
 
     assert licenses_str == expected_output
 
 
 def test_dist_package_licenses_importlib_cant_find_package(monkeypatch: pytest.MonkeyPatch) -> None:
     monkeypatch.setattr("pipdeptree._models.package.metadata", Mock(side_effect=PackageNotFoundError()))
-    dist = DistPackage(Mock(project_name="a", key="a"))
+    dist = DistPackage(Mock(project_name="a", metadata={"Name": "a"}))
     licenses_str = dist.licenses()
 
     assert licenses_str == Package.UNKNOWN_LICENSE_STR
 
 
-def test_dist_package_project_name_recovered(monkeypatch: pytest.MonkeyPatch) -> None:
-    monkeypatch.setattr("pipdeptree._models.package.metadata", lambda _: {"name": "good"})
-    foo = Mock(key="foo", project_name="bad", version="20.4.1")
-    dp = DistPackage(foo)
-    assert dp.project_name == "good"
-
-
 def test_dist_package_key_pep503_normalized() -> None:
-    foobar = Mock(key="foo.bar", project_name="foo.bar", version="20.4.1")
+    foobar = Mock(metadata={"Name": "foo.bar"}, project_name="foo.bar", version="20.4.1")
```

Comment:
There are many mock objects I see in these unit tests seem to have a `project_name` that are no longer used. Let's remove them.

Reply:
Done